core/src/main/java/software/amazon/awssdk/core/client/config/ClientAsyncConfiguration.java

```diff
@@ -83,7 +87,12 @@ public Builder toBuilder() {
 
         @Override
         public <T> Builder advancedOption(SdkAdvancedAsyncClientOption<T> option, T value) {
-            this.advancedOptions.put(option, value);
+            if (option == FUTURE_COMPLETION_EXECUTOR && value instanceof ExecutorService) {
```

Comment:
Should we check if it's closeable too and make it non-closeable? Just in case they have a weird implementation.

Reply:
+1

Reply:
Updated. Now I just wrap it in a vanilla executor, no matter what it is.

Reply:
Clever!